src/renderer/tvgInitializer.cpp

```diff
@@ -54,36 +54,30 @@ static constexpr bool operator &(CanvasEngine a, CanvasEngine b)
     return int(a) & int(b);
 }
 
-static bool _buildVersionInfo()
+static bool _buildVersionInfo(uint32_t* major, uint32_t* minor, uint32_t* micro)
 {
-    auto SRC = THORVG_VERSION_STRING;   //ex) 0.3.99
-    auto p = SRC;
+    auto VER = THORVG_VERSION_STRING;
+    auto p = VER;
     const char* x;
 
-    char major[3];
-    x = strchr(p, '.');
-    if (!x) return false;
-    memcpy(major, p, x - p);
-    major[x - p] = '\0';
+    if (!(x = strchr(p, '.'))) return false;
+    uint32_t majorVal = atoi(p);
     p = x + 1;
 
-    char minor[3];
-    x = strchr(p, '.');
-    if (!x) return false;
-    memcpy(minor, p, x - p);
-    minor[x - p] = '\0';
+    if (!(x = strchr(p, '.'))) return false;
+    uint32_t minorVal = atoi(p);
     p = x + 1;
 
-    char micro[3];
-    x = SRC + strlen(THORVG_VERSION_STRING);
-    memcpy(micro, p, x - p);
-    micro[x - p] = '\0';
+    uint32_t microVal = atoi(p);
 
     char sum[7];
-    snprintf(sum, sizeof(sum), "%s%s%s", major, minor, micro);
-
+    snprintf(sum, sizeof(sum), "%d%d%d", majorVal, minorVal, microVal);
```

Comment:
This code returns 0.14.2 and 1.4.2 the same value. and 0.3.99 may be recognized to be a higher version than 0.4.0.
I recommend to use a format that fills spaces with 0.
```suggestion
    snprintf(sum, sizeof(sum), "%02d%02d%02d", majorVal, minorVal, microVal);
```